core/src/main/java/org/apache/shiro/realm/jdbc/JdbcRealm.java

```diff
@@ -247,7 +261,11 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
             info = new SimpleAuthenticationInfo(username, password.toCharArray(), getName());
             
             if (salt != null) {
-                info.setCredentialsSalt(ByteSource.Util.bytes(salt));
+            	if (saltStyle == SaltStyle.COLUMN && saltIsBase64Encoded) {
+                    info.setCredentialsSalt(ByteSource.Util.bytes(Base64.decode(salt)));
```

Comment:
We could probably decode this once in `setSaltIsBase64Encoded()` and reduce the branch here